x-pack/plugins/security_solution/public/common/components/exceptions/add_exception_flyout/index.tsx

```diff
@@ -176,14 +176,29 @@ export const AddExceptionFlyout = memo(function AddExceptionFlyout({
 
   useEffect(() => {
     const fetchSingleDataView = async () => {
-      if (dataViewId != null && dataViewId !== '') {
-        const dv = await data.dataViews.get(dataViewId);
+      const notNullDataViewId =
+        dataViewId != null && dataViewId !== ''
+          ? dataViewId
+          : maybeRule?.data_view_id != null && maybeRule?.data_view_id !== ''
+          ? maybeRule.data_view_id
+          : null;
+      if (notNullDataViewId != null) {
+        const dv = await data.dataViews.get(notNullDataViewId);
         setIndexPattern(dv);
       }
     };
 
     fetchSingleDataView();
-  }, [data.dataViews, dataViewId, setIndexPattern]);
+  }, [data.dataViews, dataViewId, maybeRule?.data_view_id, setIndexPattern]);
+
+  const getIndexPattern = useCallback(
```

Comment:
Also here, do we really need to use `getIndexPattern` as a function and `useCallback`  here?
I saw that we invoke this function once, and exactly in this component render

Probably is simplier way to write this is - 

```
const  selectedIndexPattern= (dataViewId || maybeRule?.data_view_id) ? indexPattern
        : indexIndexPatterns
 ```